R/data_partition.R

```diff
@@ -38,21 +39,27 @@
 #' # Create multiple partitions, stratified by group - 30% equally sampled
 #' # from species in first training set, 50% in second training set and
 #' # remaining 20% equally sampled from each species in test set.
-#' out <- data_partition(iris, proportion = c(0.3, 0.5), group = "Species")
+#' out <- data_partition(iris, proportion = c(0.3, 0.5), by = "Species")
 #' lapply(out, function(i) table(i$Species))
 #'
 #' @inherit data_rename seealso
 #' @export
 data_partition <- function(data,
                            proportion = 0.7,
-                           group = NULL,
+                           by = NULL,
                            seed = NULL,
                            row_id = ".row_id",
                            verbose = TRUE,
+                           group = NULL,
                            ...) {
   # validation checks
   data <- .coerce_to_dataframe(data)
 
+  ## TODO: deprecate later
+  if (!is.null(group)) {
+    by <- group
+  }
```

Comment:
There should be a deprecation warning since the docs already say this arg is deprecated

Reply:
See my comment above. I would first submit all easystats packages to CRAN so that they work without warnings. In a later update round, we can deprecate. But I have no strong opinion on this. We can also deprecate now, meaning that once we release one package, we need to release the others as well.

Reply:
If the other packages are updated before `datawizard` is released then they will error because the arg `by` won't exist yet in the CRAN `datawizard`. So `datawizard` will have to be updated first anyway

Reply:
Ok I understand

Reply:
At the same time, when we submit `datawizard` we could just tell the CRAN team that most of the revdepcheck warnings are in packages in the same organization and that updates are coming in the few days after

Reply:
I don't mind deprecating now. The "FIXME" comment in `data_xtabulate.R`, however, requires _insight_ to be on CRAN first, and I just submitted that package this week. But we can wait with _datawizard_ anyway?

Reply:
Yes no problem